File: server/tests/fixtures/alpha_rule_triggers/javascript/arch_external_integration.js
// Should trigger: arch.external_integration
// Fetch API - external HTTP call
async function fetchUserData(userId) {
    const response = await fetch(`https://api.example.com/users/${userId}`);
    return response.json();
}

// XMLHttpRequest - external call
function loadData(url) {
    const xhr = new XMLHttpRequest();
    xhr.open('GET', url);
    xhr.send();
}

// Axios HTTP client
const axios = require('axios');

async function postData(data) {
    const result = await axios.post('/api/submit', data);
    return result.data;
}

File: server/tests/fixtures/alpha_rule_triggers/csharp/arch_external_integration.cs
// Should trigger: arch.external_integration
using System.Net.Http;
using System.Data.SqlClient;

namespace MyApp.Services
{
    // HTTP client - external integration
    public class ExternalService
    {
        private readonly HttpClient _httpClient = new HttpClient();

        public async Task<string> FetchUserDataAsync(string userId)
        {
            var response = await _httpClient.GetAsync($"https://api.example.com/users/{userId}");
            return await response.Content.ReadAsStringAsync();
        }

        // Database connection - external integration
        public SqlConnection GetDatabaseConnection()
        {
            return new SqlConnection("Server=localhost;Database=app;User Id=user;Password=pass;");
        }
    }
}
